src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -162,6 +162,22 @@ Plugin::Plugin() {
     m_compiled_model_runtime_properties["OV_VERSION"] = ov_version.buildNumber;
 }
 
+void Plugin::set_cache_info(const std::shared_ptr<const ov::Model>& model, ExecutionConfig& config) const {
+    // GPU_WEIGHTS_PATH is used for the weightless cache mechanism which is used only with
+    // ov::CacheMode::OPTIMIZE_SIZE setting. Not setting GPU_WEIGHTS_PATH will result in not
+    // using that mechanism.
```

Comment:
I think it should be 'WEIGHTS_PATH', not 'GPU_WEIGHTS_PATH'.
I'd also consider to rename property 'WEIGHTS_PATH' to 'MODEL_BIN_FILE_PATH'.